ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+        row1=rgrdf1['row'][:]
+        col1=rgrdf1['col'][:]
+        frac_b1=rgrdf1['frac_b'][:]
+        n_s1=len(S1)
+
+        rgrdf2=nc.Dataset(wgtsfile2)
+        S2=rgrdf2['S'][:].copy()
+        row2=rgrdf2['row'][:]
+        col2=rgrdf2['col'][:]
+        frac_b2=rgrdf2['frac_b'][:]
+        n_s2=len(S2)
+
+        rgrdf3=nc.Dataset(wgtsfile3)
+        S3=rgrdf3['S'][:].copy()
+        row3=rgrdf3['row'][:]
+        col3=rgrdf3['col'][:]
+        frac_b3=rgrdf3['frac_b'][:]
+        n_s3=len(S3)
+
+        rgrdf4=nc.Dataset(wgtsfile4)
+        S4=rgrdf4['S'][:].copy()
+        row4=rgrdf4['row'][:]
+        col4=rgrdf4['col'][:]
+        frac_b4=rgrdf4['frac_b'][:]
+        n_s4=len(S4)
+
+        #Interpolate landmask to rectilinear grid 
+        if model == 'MOM' or model == 'CICE':
+            t1_start = process_time()
+            print('beginning the regridding for mask2d') 
+            if regridtype == 'manual':
+                rgmask2d=esmfmanualregrid(mask2d,S1,row1,col1,frac_b1,n_s1,'bilinear',dimyh,dimxh,len(latd),len(lond))
+            t1_stop = process_time()
+            print("Remapped Elapsed time:", t1_stop-t1_start) 
+            print('Data has been regridded for mask2d')
+            rgmask2d = np.where(rgmask2d>0.0,  1.0, 0.0)
+
+        if model == 'MOM':
+            print('beginning the regridding for mask3d')                                                                                                                                                                                                 
+            if regridtype == 'manual':
+               rgmask3d=esmfmanualregrid(mask3d,S1,row1,col1,frac_b1,n_s1,'bilinear',dimyh,dimxh,len(latd),len(lond))
+            print('Data has been regridded for mask3d') 
+            rgmask3d = np.where(rgmask3d>0.0,  1.0, 0.0)       
+
+        #Create Mask NETCDF File#########      
+        testfile='masks_'+dstgrds[jj]+'.nc'
+        os.system('rm -vf '+testfile)
+        f = nc.Dataset(testfile,'w', format='NETCDF4') 
+
+        if model == 'MOM':
+            f.createDimension('z_l', nlevs) 
+        f.createDimension('lon', len(lond))
+        f.createDimension('lat', len(latd))
+        f.createDimension('time', None)
+
+        if model == 'MOM':
+            depth = f.createVariable('z_l', 'f4', 'z_l')
+            depth.long_name='Depth at cell center'
+            depth.units = 'meters'
+
+        longitude = f.createVariable('lon', 'f4', 'lon')
+        longitude.long_name='longitude'
+        longitude.units = 'degrees_east'
+
+        latitude = f.createVariable('lat', 'f4', 'lat')  
+        latitude.long_name = 'latitude'
+        latitude.units = 'degrees_north'
+
+        rgmask2dn = f.createVariable('rgmask2d1', 'f4', ('time', 'lat', 'lon'))
+        if model == 'MOM':
+            rgmask3dn = f.createVariable('rgmask3d', 'f4', ('time', 'z_l','lat', 'lon')) 
+        time = f.createVariable('Time', 'i4', 'time')
+
+        longitude[:] = lond    
```

Comment:
these are things you would do in a Fortran code.
This is not proper use of the python language.